### PR TITLE
Remove -c option from udp endpoints in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR from the undocumented "-c" option from udp endpoints in C#.

This was a server-endpoint option. For client option, the behavior is always "connect".

### CODE REVIEW

File: csharp/src/Ice/Internal/UdpTransceiver.cs [hunk 1]
```diff
@@ -726,12 +692,11 @@ internal UdpTransceiver(
...
         string host,
         int port,
-        string mcastInterface,
-        bool connect)
+        string mcastInterface)
     {
         _endpoint = endpoint;
         _instance = instance;
-        _state = connect ? StateNeedConnect : StateNotConnected;
+        _state = StateNotConnected;
```
Comment (bernardnormier) [inline csharp/src/Ice/Internal/UdpTransceiver.cs:699]: A server transceiver (incoming = true) starts in the StateNotConnected state and remains in that state.
A client transceiver (incoming = false) starts in the StateNeedConnect state and goes up to StateConnected. It never reaches StateNotConnected.

File: csharp/src/Ice/Internal/UdpTransceiver.cs [hunk 2]
```diff
@@ -554,8 +518,9 @@ public ConnectionInfo getInfo(bool incoming, string adapterName, string connecti
         {
             EndPoint localEndpoint = Network.getLocalAddress(_fd);
 
-            if (_state == StateNotConnected) // TODO: eliminate all these states
+            if (_state == StateNotConnected) // a server endpoint
             {
+                Debug.Assert(incoming);
```
Comment (bernardnormier) [inline csharp/src/Ice/Internal/UdpTransceiver.cs:523]: Question: do we create a new Connection and potentially ConnectionInfo for each incoming UDP request?

Comment (pepone) [reply csharp/src/Ice/Internal/UdpTransceiver.cs:523]: Looking at the IncomingConnectionFactory I believe there is a single udp Connection created by the factory constructor. But then caching the info object wouldn't be correct.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

File: csharp/src/Ice/Internal/UdpTransceiver.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/src/Ice/Internal/UdpTransceiver.cs:523]: I believe we create a new connection for each request.  And udp bi-dir relies on these per-request connections:
https://github.com/zeroc-ice/ice/blob/41307f90a6d1ebe097f3aef5b5b5e802dd78f543/csharp/test/Ice/udp/AllTests.cs#L200

**Pull request merged**